vmicore/src/lib/os/windows/KernelOffsets.cpp

```diff
@@ -45,10 +46,24 @@ namespace VmiCore::Windows
                                 .Right = vmiInterface->getKernelStructOffset("_RTL_BALANCED_NODE", "Right")},
             .fileObject = {.FileName = vmiInterface->getKernelStructOffset("_FILE_OBJECT", "FileName")},
             .section = {.controlArea = vmiInterface->getKernelStructOffset("_SECTION", "u1")},
-            .kprocess = {.DirectoryTableBase = vmiInterface->getKernelStructOffset("_KPROCESS", "DirectoryTableBase")},
+            .kprocess = {.directoryTableBase = vmiInterface->getKernelStructOffset("_KPROCESS", "DirectoryTableBase"),
+                         .userDirectoryTableBase =
+                             vmiInterface->getKernelStructOffset("_KPROCESS", "DirectoryTableBase")},
             .subSection = {.ControlArea = vmiInterface->getKernelStructOffset("_SUBSECTION", "ControlArea")},
             .exFastRef = {.Object = vmiInterface->getKernelStructOffset("_EX_FAST_REF", "Object")}};
 
+        if (vmiInterface->getWindowsBuild() >= winBuildRedstone3)
+        {
+            // Will fail for Windows 10 versions below "Win 10 1803 Redstone 4"
+            auto userDTB = vmiInterface->getKernelStructOffset("_KPROCESS", "UserDirectoryTableBase");
+
+            // KPTI is implemented but not activated
+            if (userDTB != 0)
```

Comment:
What? The offset should always be the correct one. You have to read the value at this offset and check if it's zero.

Reply:
Don't know why I did this :>